components/style/gecko_bindings/mod.rs

```diff
@@ -2,16 +2,16 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#[allow(dead_code, non_camel_case_types)]
+#[allow(dead_code, improper_ctypes, non_camel_case_types)]
```

Comment:
Hmm... Please, add a comment here saying that we can't rely on this warning because of `PhantomData`, and that we should remove it once the lint ignores it.